FileSystem/FileMapperMocker.js

```diff
@@ -10,28 +25,490 @@ var USE_FILEMAPPER_MOCKER = true;
+    var folder = DriveApp.getFolderById(folderId);
+    var destination = targetFolder.createFolder(folder.getName());
+    copyFolderUtil(folder, destination);
+  }
+}
+
+/**
+ * Recursively copy source folder to target folder
+ * @param {Folder} source Source Folder
+ * @param {Folder} target Target Folder
+ */
+function copyFolderUtil(source, target) {
+  var folders = source.getFolders();
+  var files = source.getFiles();
+  while (files.hasNext()) {
+    var file = files.next();
+    file.makeCopy(file.getName(), target);
+  }
+  while (folders.hasNext()) {
+    var subFolder = folders.next();
+    var folderName = subFolder.getName();
+    var targetFolder = target.createFolder(folderName);
+    copyFolderUtil(subFolder, targetFolder);
+  }
+}
+
+/**
+ * Find files by pattern API
+ * This API searches for files which match the given pattern consisting
+ * of wildcards. Wildcard ? matches with any one valid character.
+ * Wildcard * matches with one or more valid characters
+ * @param {string} localPath Local file path of the file
+ * @return {Array} List of files matching the pattern
+ */
+function findFilesByPattern(localPath) {
+  // Get platform agnostic relative path split
+  var pathSplit = this.getRelativePathSplit(localPath);
+  // Get base directory
+  var root = this.getBaseDriveFolder();
+  // Get parent directory of file
+  for (var i = 0; i < pathSplit.length - 1; i++) {
+    var folderIterator = root.getFoldersByName(pathSplit[i]);
+    if (!folderIterator.hasNext()) {
+      return [];
+    }
+    root = folderIterator.next();
+  }
+  var pattern = pathSplit[pathSplit.length - 1];
+  // Generate regex expression
+  var patternRegex = generateRegexFromPattern(pattern);
+  var fileIterator = root.getFiles();
+  var matches = [];
+  // Search for files matching pattern
+  while (fileIterator.hasNext()) {
+    var file = fileIterator.next();
+    if (patternRegex.test(file.getName())) {
+      matches.push(file.getName());
+    }
+  }
+  return matches;
+}
+
+/**
+ * Find folders by pattern API
+ * This API searches for folders which match the given pattern consisting
+ * of wildcards. Wildcard ? matches with any one valid character.
+ * Wildcard * matches with one or more valid characters
+ * @param {string} localPath Local file path of the folder
+ * @return {Array} List of folders matching the pattern
+ */
+function findFoldersByPattern(localPath) {
+  // Get platform agnostic relative path split
+  var pathSplit = this.getRelativePathSplit(localPath);
+  // Get base directory
+  var root = this.getBaseDriveFolder();
+  // Get parent directory of file
+  for (var i = 0; i < pathSplit.length - 1; i++) {
+    var folderIterator = root.getFoldersByName(pathSplit[i]);
+    if (!folderIterator.hasNext()) {
+      return [];
+    }
+    root = folderIterator.next();
+  }
+  // Generate regex matching pattern
+  var pattern = pathSplit[pathSplit.length - 1];
+  var patternRegex = generateRegexFromPattern(pattern);
```

Comment:
All code in the method till here are same as the one for the function above, extract out a function

Reply:
Fixed: Refactored to using findByPatternUtil()